server/models/Earnings.js

```diff
@@ -152,8 +155,8 @@ class Earnings {
 
     if (sortBy === 'grower') {
       formattedEarnings.sort((a, b) => {
-        const nameA = a.grower?.toUpperCase(); // ignore upper and lowercase
-        const nameB = b.grower?.toUpperCase(); // ignore upper and lowercase
+        const nameA = a.grower?.toUpperCase() || ""; // ignore upper and lowercase
+        const nameB = b.grower?.toUpperCase() || "";  // ignore upper and lowercase
```

Comment:
I don't think we need to add this the grower names should not be empty @dadiorchen